### PR TITLE
chore: feature flag for swap zero fee

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

[Swap fee=zero](https://github.com/cowprotocol/cowswap/pull/3469) feature is hidden under `swapZeroFee` flag

### CODE REVIEW

File: apps/cowswap-frontend/src/legacy/utils/trade.ts [hunk 1]
```diff
@@ -129,7 +135,7 @@ export function getSignOrderParams(params: PostOrderParams): SignOrderParams {
       buyAmount,
       validTo,
       appData: appData.appDataKeccak256,
-      feeAmount: '0',
+      feeAmount: (swapZeroFee ? '0' : feeAmount?.quotient.toString()) || '0',
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/utils/trade.ts:138]: It defaults to the behavior matching when the feature flag is on?

Comment (shoom3301) [reply apps/cowswap-frontend/src/legacy/utils/trade.ts:138]: It's actually unreal case when `feeAmount` is falsy. This is a part of legacy code

**Review by alfetopito — APPROVED** [review]
The flag works as advertised!

I think I found an issue with BUY orders, but it's unrelated to this changed, so I'm approving it.

File: apps/cowswap-frontend/src/common/containers/FeatureGuard/index.tsx [hunk 2]
```diff
@@ -5,14 +5,15 @@ import { useFeatureFlags } from 'common/hooks/featureFlags/useFeatureFlags'
...
-export function FeatureGuard({ featureFlag, children }: FeatureGuardProps) {
+export function FeatureGuard({ featureFlag, children, defaultContent }: FeatureGuardProps) {
   const flags = useFeatureFlags()
 
   if (flags[featureFlag]) {
     return <>{children}</>
   }
 
-  return null
+  return defaultContent || null
```
Comment (anxolin) [inline apps/cowswap-frontend/src/common/containers/FeatureGuard/index.tsx:18]: cool

**Comment by anxolin** [comment]
Works like a charm!

I tested both ON/OFF the flags

**Pull request merged**